gammapy/catalog/lhaaso.py

```diff
@@ -0,0 +1,190 @@
+            model.parameters[name].error = value
+
+        return model
+
+    @staticmethod
+    def _get_components_geom(models):
+        energy_axis = MapAxis.from_energy_bounds(
+            "0.1 TeV", "2000 TeV", nbin=10, per_decade=True, name="energy"
+        )
+        regions = [m.spatial_model.evaluation_region for m in models]
+        geom = RegionGeom.from_regions(
+            regions, binsz_wcs="0.05 deg", axes=[energy_axis]
+        )
+        return geom.to_wcs_geom()
+
+    def sky_model(self, which="both"):
+        """Sky model (`~gammapy.modeling.models.SkyModel`).
+
+        * ``which="both"`` - Use first model or create a composite template if both models are available.
+        * ``which="KM2A"`` - Sky model for KM2A analysis if available.
+        * ``which="WCDA"`` - Sky model for WCDA analysis if available.
+
+        """
+
+        if which == "both":
+            wcda = self.sky_model(which="WCDA")
+            km2a = self.sky_model(which="KM2A")
+            models = [m for m in [wcda, km2a] if m is not None]
+            if len(models) == 2:
+                geom = self._get_components_geom(models)
+                mask = geom.energy_mask(energy_max=25 * u.TeV)
+                geom = geom.as_energy_true
+                wcda_map = Models(wcda).to_template_sky_model(geom).spatial_model.map
+                model = Models(km2a).to_template_sky_model(geom, name=km2a.name)
+                model.spatial_model.map.data[mask] = wcda_map.data[mask]
+                model.spatial_model.filename = f"{model.name}.fits"
+                return model
+            else:
+                return models[0]
+        else:
+            _, is_ul = self._parse("N0", which)
+            if is_ul:
+                return None
+            else:
+                return SkyModel(
+                    spatial_model=self.spatial_model(which),
+                    spectral_model=self.spectral_model(which),
+                    name=self.name,
+                )
+
+
+class SourceCatalog1LHAASO(SourceCatalog):
+    """First LHAASO catalog.
+
+    One source is represented by `~gammapy.catalog.SourceCatalogObject3HWC`.
+
+    The data is from table 1 in the paper [1]_.
+
+    The catalog table contains 90 rows / sources.
+
+    References
+    ----------
+    .. [1] 1LHAASO: The First LHAASO Catalog of Gamma-Ray Sources,
+       <https://ui.adsabs.harvard.edu/abs/2023arXiv230517030C/abstract>`__
+    """
+
+    tag = "1LHAASO"
+    """Catalog name"""
+
+    description = "1LHAASO catalog from the LHAASO observatory"
+    """Catalog description"""
+
+    source_object_class = SourceCatalogObject1LHAASO
+
+    def __init__(self, filename="$GAMMAPY_DATA/catalogs/1LHAASO_catalog.fits"):
+        table = Table.read(make_path(filename))
+
+        source_name_key = "Source_Name"
+
+        super().__init__(table=table, source_name_key=source_name_key)
+
+    def to_models(self, **kwargs):
+        """Create Models object from catalogue"""
```

Comment:
Would be good to expose `which` option here

Reply:
done